From f6c22220c7da8123773306b82822861366af0ef1 Mon Sep 17 00:00:00 2001
From: Long Dang <long.dang.aj@rvc.renesas.com>
Date: Thu, 18 Aug 2016 16:57:22 +0700
Subject: [PATCH 1/3] gst145-omxbufferpool: fix memory leak in
 gst_omx_buffer_pool_request_videosink_buffer_creation function

Porting from a patch in Linux 3.10 environment made by Tam Nguyen <tamnm2@fsoft.com.vn>

Signed-off-by: Long Dang <long.dang.aj@rvc.renesas.com>
---
 omx/gstomxbufferpool.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/omx/gstomxbufferpool.c b/omx/gstomxbufferpool.c
index 98e5526..a361eac 100644
--- a/omx/gstomxbufferpool.c
+++ b/omx/gstomxbufferpool.c
@@ -580,6 +580,9 @@ gst_omx_buffer_pool_request_videosink_buffer_creation (GstOMXBufferPool * pool,
 
   if (!gst_pad_peer_query (GST_VIDEO_DECODER_SRC_PAD (pool->element), query)) {
     GST_ERROR_OBJECT (pool, "videosink_buffer_creation_request query failed");
+    gst_query_unref (query);
+    g_array_free (dmabuf_array, TRUE);  /* free array when function error or failed */
+    g_array_free (stride_array, TRUE);  /* free array when function error or failed */
     return NULL;
   }
 
@@ -588,6 +591,9 @@ gst_omx_buffer_pool_request_videosink_buffer_creation (GstOMXBufferPool * pool,
   if (buffer == NULL) {
     GST_ERROR_OBJECT (pool,
         "could not get a buffer from videosink_buffer_creation query");
+    gst_query_unref (query);
+    g_array_free (dmabuf_array, TRUE);  /* free array when function error or failed */
+    g_array_free (stride_array, TRUE);  /* free array when function error or failed */
     return NULL;
   }
 
-- 
1.9.1

